src/language_features/kak_lsp_custom.rs

```diff
@@ -0,0 +1,318 @@
+                    },
+                    position: get_lsp_position(&meta.buffile, &symbol_position, ctx).unwrap(),
+                },
+                work_done_progress_params: Default::default(),
+            };
+
+            // This context is shown at the top of the modal
+            let context = format!(
+                                "line {}:{}:{{+b@KindAndName}}{:?} {}{{KindAndName}} (Press `g` to goto this position. Press any other key to continue)",
+                                symbol_position.line, symbol_position.column, kind, escape_brace(&name)
+                            );
+
+            // This kak script is appended after the kakoune `lsp-show-hover`
+            let do_after = format!(
+                "on-key %[eval %sh[
+                     if [ \"$kak_key\" = \"g\" ];
+                     then echo 'info -style modal
+                                exec {}g{}lh
+                                ';
+                     else echo 'info -style modal';
+                     fi
+                 ]]",
+                symbol_position.line, symbol_position.column
+            );
+
+            // Make a HoverRequest to the backend language server and then call `hover::editor_hover()`
+            ctx.call::<HoverRequest, _>(
+                meta,
+                req_params,
+                move |ctx: &mut Context, meta, result| {
+                    editor_hover(
+                        meta,
+                        Some(HoverModal { context, do_after }),
+                        PositionParams {
+                            position: symbol_position,
+                        },
+                        result,
+                        ctx,
+                    )
+                },
+            );
+        } else {
+            let location = format!(
+                "%§{}§ {} {}",
+                filename.replace("§", "§§"),
+                symbol_position.line,
+                symbol_position.column
+            );
+            let command = format!("edit! -existing {}", location);
+            ctx.exec(meta, command);
+        }
+    } else {
+        ctx.exec(
+            meta,
+            "eval %[
+                info -style modal 'Not found!\n\nPress any key to continue'
+                on-key %[info -style modal]
+            ]",
+        );
+    }
+}
+
+/// Does the currently being inspected symbol's location "exceed" the
+/// current cursor location?
+fn exceeds(
+    symbol_position: KakounePosition,
+    symbol_kind: SymbolKind,
+    cur_position: KakounePosition,
+    search_next: bool,
+    expected_kind: &str,
+) -> bool {
+    // expected kind is the symbol for which the user is searching for
+    // e.g. `Function`, `Constructor` etc.
+    if !expected_kind.is_empty() && format!("{:?}", symbol_kind) != expected_kind {
+        return false;
+    }
+
+    // If searching forwards, the first element that has a greater line/column
+    // combination
```

Comment:
please reflow to 100 columns (which should be the rustfmt default). Again, I'll add the editorconfig soon

Reply:
Done